pkg/core/blockchain.go

```diff
@@ -1622,7 +1625,7 @@ func (bc *Blockchain) storeBlock(block *block.Block, txpool *mempool.Pool) error
 				stop = start + 1
 			}
 			for index := start; index < stop; index++ {
-				err := kvcache.DeleteBlock(bc.GetHeaderHash(index))
+				err := kvcache.DeleteBlock(bc.GetHeaderHash(index), bc.config.Ledger.RemoveUntraceableHeaders)
```

Comment:
The problem is that this code will remove untraceable headers, whereas GC is not aware of it. In subsequent GC round the node will try to `removeOldTransfers`, and this attempt will fail because we don't have old headers by this moment.

Reply:
Oh, turns out that I reviewed only first commit from the batch. This problem is fixed in the last commit.